### PR TITLE
Feat/nginx

### PR DESCRIPTION
**Pull request by MichaelLukowski**
Description about what this pull request does.

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### New Features
Changing from apache 2 to nginx
updating to uwsgi

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,27 @@
-FROM ubuntu:16.04
+# To run: docker run -v /path/to/wsgi.py:/var/www/indexd/wsgi.py --name=indexd -p 81:80 indexd
+# To check running container: docker exec -it indexd /bin/bash 
+
+FROM quay.io/cdis/py27base:pybase2-1.0.0
```
Comment (frickjack) [inline Dockerfile]: use `pybase2-1.0.1` : https://github.com/uc-cdis/cloud-automation/releases

File: Dockerfile [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
 RUN COMMIT=`git rev-parse HEAD` && echo "COMMIT=\"${COMMIT}\"" >indexd/index/version_data.py
 RUN VERSION=`git describe --always --tags` && echo "VERSION=\"${VERSION}\"" >>indexd/index/version_data.py
 RUN python setup.py install
 
-RUN mkdir -p /var/www/indexd/ && chmod 777 /var/www/indexd && cp /indexd/wsgi.py /var/www/indexd/wsgi.py && cp /indexd/bin/indexd /var/www/indexd/indexd
-
-#
-# Custom apache2 logging - see http://www.loadbalancer.org/blog/apache-and-x-forwarded-for-headers/
-#
-COPY ./deployment/apache-indexd.conf /etc/apache2/sites-available/apache-indexd.conf
```
Comment (fantix) [inline Dockerfile]: Remove `deployment/apache-indexd.conf` too perhaps?

File: Dockerfile [hunk 3]
```diff
@@ -1,31 +1,27 @@
...
-
-RUN a2ensite apache-indexd
-RUN a2enmod headers
-RUN a2enmod reqtimeout
-RUN a2dissite 000-default.conf
 
 EXPOSE 80
 
 WORKDIR /var/www/indexd
+RUN cp /indexd/wsgi.py .
```
Comment (fantix) [inline Dockerfile]: Is it okay to point the `wsgi-file` in `uwsgi.ini` to `/indexd/wsgi.py` directly?

File: Dockerfile (hunk 2 above)
Comment (fantix) [reply Dockerfile]: Looking into that file, it seems we missed out [the Origin Access Control header](https://github.com/uc-cdis/indexd/commit/b5b3114a1fabcbcef848f823bbf1cec45f66bcfd#diff-3254677a7917c6c01f55212f86c57fbfR17) in the new image - just a reminder here, we probably added that in the `revproxy` service already.

File: deployment/uwsgi/uwsgi.ini [hunk 4]
```diff
@@ -0,0 +1,29 @@
+[uwsgi]
+protocol = uwsgi
+socket = /var/run/gen3/uwsgi.sock
+buffer-size = 32768
+chmod-socket = 666
+master = true
+processes = 2
```
Comment (fantix) [inline deployment/uwsgi/uwsgi.ini]: A failing database schema migration may prevent one of the worker processes from importing the WSGI application correctly, leading to a situation with 1 functional process and 1 broken one. I'm still thinking about how to properly handle that.

File: Dockerfile (hunk 3 above)
Comment (MichaelLukowski) [reply Dockerfile]: Moved the location `uwsgi.ini` points to `wsgi.pi`

File: Dockerfile (hunk 2 above)
Comment (MichaelLukowski) [reply Dockerfile]: I removed this file

**Pull request merged**